orchagent/watermarkorch.cpp

```diff
@@ -98,7 +98,10 @@ void WatermarkOrch::handleWmConfigUpdate(const std::string &key, const std::vect
         {
             if (i.first == "interval")
             {
-                auto intervT = timespec { .tv_sec = to_uint<uint32_t>(i.second.c_str()) , .tv_nsec = 0 };
+                struct tm tmInterval;
+                memset(&tmInterval, 0, sizeof(tmInterval));
+                tmInterval.tm_sec = to_uint<uint32_t>(i.second.c_str());
```

Comment:
>uint32_t [](start = 44, length = 8)

`tm_sec` is `int`, why not convert to `int` directly? You may check the validity after conversion.

Reply:
Should we really need to do mktime(),
For example,
If tm_sec value is 5899989  then mktime returns Time_t 18446744071506376405 which is different from the earlier functionality, where polling count was directly used as value for seconds.

Reply:
Reverted the mktime() changes and test passes.